handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -1109,7 +1122,9 @@ public final synchronized void closeOutbound() {
 
     @Override
     public final synchronized boolean isOutboundDone() {
-        return isOutboundDone;
+        // Check if there is anything left in the outbound buffer.
+        // We need to ensure we only call SSL.pendingWrittenBytesInBIO(...) if the engine was not destroyed yet.
+        return outboundClosed && (isDestroyed() || SSL.pendingWrittenBytesInBIO(networkBIO) == 0);
```

Comment:
https://github.com/netty/netty/pull/6090#discussion_r90548328 was collapsed but still has some discussion

me:

> is checking `isDestroyed()` necessary? Setting the `destroyed` variable to "true" also means that `outboundClosed` will be set to `true`.

@normanmaurer 

> yes as we only want to call `SSL.pendingWrittenBytesInBIO(...)` if the engine was not destroyed before. If we not check this it may result in a segfault as the SSL pointer was freed before.

me: 

> but `isDestroyed() == true` means that `outboundClosed == true`, and if this is not the case then we have a bug. Am I missing something?